tests/run/independent_spanning_type.pyx

```diff
@@ -0,0 +1,69 @@
+from cython cimport typeof
+
+ctypedef bint mybool
+
+cdef mybool mybul = True
+cdef bint bul = True
+cdef int num = 42
+
+
+def CondExprNode_to_obj(test):
+    """
+    >>> CondExprNode_to_obj(True)
+    True
+    2
+    >>> CondExprNode_to_obj(False)
+    True
+    84
+    """
+
+    print(typeof(mybul if test else num) == typeof(bul if test else num) == typeof(object))
+
+    return (mybul if test else num) + (bul if test else num)
+
+
+def BoolBinopNode_to_obj():
+    """
+    >>> BoolBinopNode_to_obj()
+    True
+    2
+    """
+
+    print(typeof(mybul or num) == typeof(bul or num) == typeof(object))
+
+    return (mybul or num) + (bul or num)
+
+
+cdef int test_bool(mybool arg):
+    return <int>arg
+
+
+def CondExprNode_to_bool(test):
+    """
+    >>> CondExprNode_to_bool(True)
+    True
+    0
+    >>> CondExprNode_to_bool(False)
+    True
+    2
+    """
+
+    print(typeof(not mybul if test else mybul) == typeof(not bul if test else bul) == typeof(bul))
+
+    # test_bool would silently crash if one of the types is casted
+    # to Python object and not just assigned.
+    # It happens when type is wrongly inferred to Python object
+    # and not bin or mybool.
```

Comment:
```suggestion
    # test_bool() would silently crash if one of the types is cast
    # to Python object and not just assigned.
    # It happens when a type is wrongly inferred as Python object
    # instead of bint or mybool.
```